two_factor/views/core.py

```diff
@@ -700,6 +705,99 @@ def get_context_data(self):
         }
 
 
+@class_view_decorator(never_cache)
+@class_view_decorator(login_required)
+class ResetSetupGeneratorView(IdempotentSessionWizardView):
+    """
+    View for changing the two-factor authentication method from phone number to token generator.
+    """
+    template_name = 'two_factor/core/setup_reset_generator.html'
+    success_url = settings.LOGIN_REDIRECT_URL
+    qrcode_url = 'two_factor:qr'
+    session_key_name = 'django_two_factor-qr_secret_key'
+
+    form_list = (
+        ('generator', TOTPDeviceForm),
```

Comment:
There's only one form here, what about yubikey and phone methods?

Reply:
This is exactly the point: this view is used when you already have a phone as double authentication method and want to go to the generator.

Reply:
What if I want to change to another method?

Reply:
Well then you have the other view that allows you to go from the token generator to the SMS/call. You can only have one method for double authentication at a time. If you previously were using the google authenticator, you'll arrive on the view to use SMS/call. If you previously were using the SMS/call, you'll arrive on the view to use google authenticator. If you want to add extra backup tokens or phone etc nothing changes. All of this assumes you already have enable the double authentication method. If not, then you are still pointing to the initial setup view.

Reply:
And what about yubikey?

Reply:
I will add this one. My bad.

Reply:
tests are added, the case of yhe use of a yubikey is also considered